From 848dd378bb7fa53be7b88016335fa89d520f2ea3 Mon Sep 17 00:00:00 2001
From: Colin Guthrie <cguthrie@mandriva.org>
Date: Sat, 16 Oct 2010 12:26:32 +0100
Subject: [PATCH 2/2] intended-roles: Mark devices with a form factor of 'headset' as being appropriate for 'phone' streams

This was a result a report by Patrick Ben Koetter relating to his
Logitech Wireless Headset.
---
 src/pulsecore/sink.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/src/pulsecore/sink.c b/src/pulsecore/sink.c
index 8d66f98..a446f02 100644
--- a/src/pulsecore/sink.c
+++ b/src/pulsecore/sink.c
@@ -2686,7 +2686,8 @@ pa_bool_t pa_device_init_intended_roles(pa_proplist *p) {
         return TRUE;
 
     if ((s = pa_proplist_gets(p, PA_PROP_DEVICE_FORM_FACTOR)))
-        if (pa_streq(s, "handset") || pa_streq(s, "hands-free")) {
+        if (pa_streq(s, "handset") || pa_streq(s, "hands-free")
+            || pa_streq(s, "headset")) {
             pa_proplist_sets(p, PA_PROP_DEVICE_INTENDED_ROLES, "phone");
             return TRUE;
         }
-- 
1.7.1

